components/AudioPlayer/useAudio.ts

```diff
@@ -9,19 +9,22 @@ export function useAudio(audioSrc: string) {
   const [speed, setSpeed] = useState(speeds[0]);
   const intervalRef = useRef<number | null>(null);
 
-  useEffect(function () {
-    setAudio(new Audio(audioSrc));
-  }, []);
+  useEffect(
+    function () {
+      setAudio(new Audio(audioSrc));
```

Comment:
Wydaje mi się, że jak zrobisz taki return, to audio przestanie grać jak zmienisz stronę.

```suggestion
      setAudio(new Audio(audioSrc));
      
      return () => setAudio(undefined);
```

Reply:
Nie pomogło